components/shared/DayRangePicker.tsx

```diff
@@ -0,0 +1,45 @@
+import { useRouter } from "next/router";
+import SingleSelect from "components/atoms/Select/single-select";
+
+export type DayRange = 7 | 30 | 90;
+
+const dayRanges: { label: `${DayRange}d`; value: `${DayRange}` }[] = [
+  {
+    label: "7d",
+    value: "7",
+  },
+  {
+    label: "30d",
+    value: "30",
+  },
+  {
+    label: "90d",
+    value: "90",
+  },
+];
+
+interface DayRangePickerProps {
+  onDayRangeChanged?: (value: string) => void;
+}
+
+export const DayRangePicker = ({ onDayRangeChanged }: DayRangePickerProps) => {
+  const router = useRouter();
+  const dayRange = (Number(router.query.range) || 30) as DayRange;
+
+  const onDefaultDayRangeChanged = (value: string) => {
+    router.push({ query: { ...router.query, range: value } });
+    onDayRangeChanged?.(value);
```

Comment:
I made this an optional prop as the component handles all the logic for routing/querystring, but someone might still want to do something. Maybe we don't even need this atm because maybe [YAGNI](https://codibly.com/news-insights/yagni-how-to-do-things-when-you-actually-need-them-to-be-done/).